src/core/bittorrent/torrenthandle.cpp

```diff
@@ -1812,7 +1822,8 @@ void TorrentHandle::flushCache()
 
 QString TorrentHandle::toMagnetUri() const
 {
-    return m_torrentInfo.toMagnetUri();
+    if (!isValid()) return QString();
```

Comment:
From the docs: `If the torrent handle is invalid, an empty string is returned.`
So you don't need this.